core/domain/exp_domain.py

```diff
@@ -2787,6 +2787,12 @@ def generate_content_id(self, content_id_prefix):
                 )
             ]
 
+            all_valid_ca_keys = [ca_spec.name for ca_spec in ca_specs]
+            keys_to_remove = [
+                key for key in ca_dict if key not in all_valid_ca_keys]
+            for key in keys_to_remove:
+                del ca_dict[key]
```

Comment:
I think this might be clearer without the list:

```suggestion
            for key in ca_dict:
                if key not in all_valid_ca_keys:
                    del ca_dict[key]
```

Reply:
That was actually what I originally tried :P It doesn't work, it gives a runtime error saying something like "dict modified while executing for loop".

Reply:
```suggestion
            for key in list(ca_dict.keys()):
                if key not in all_valid_ca_keys:
                    del ca_dict[key]
```

Should do the trick, then.

Reply:
Hm, I don't see what the issue is with the current implementation? I do like the idea conceptually of not modifying the thing I'm iterating over, so I actually prefer the existing code to the new suggestion.

Do you feel strongly about this? If so, I'd be happy to ask around in case I have a blind spot here.

Reply:
Nope not a big deal, let's disregard.